File: wifiLib/src/main/java/com/duke/wifilib/WifiReceiver.java
package com.duke.wifilib;

import android.content.Context;
import android.content.Intent;
import android.net.wifi.WifiManager;
import android.text.TextUtils;

/**
 * @Author: duke
 * @DateTime: 2020-05-17 11:20
 * @Description:
 */
public class WifiReceiver extends BaseReceiver {
    private static final String WIFI_ACTION = WifiManager.SCAN_RESULTS_AVAILABLE_ACTION;
    private OnScanCompletedListener onScanCompletedListener;

    public interface OnScanCompletedListener {
        void onScanCompleted(boolean isSuccess);
    }

    public void register(Context context, OnScanCompletedListener onScanCompleted) {
        super.register(context, WIFI_ACTION);
        onScanCompletedListener = onScanCompleted;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        boolean success = false;
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            success = intent.getBooleanExtra(WifiManager.EXTRA_RESULTS_UPDATED, false);
        }
        String action = intent.getAction();
        if (!TextUtils.isEmpty(action)
                && WIFI_ACTION.equals(action)
                && onScanCompletedListener != null) {
            onScanCompletedListener.onScanCompleted(success);
        }
    }
}

File: wifiLib/src/main/java/com/duke/wifilib/WifiReceiver.kt
package com.duke.wifilib

import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager
import android.os.Build
import android.text.TextUtils

/**
 * @Author: duke
 * @DateTime: 2020-05-17 11:20
 * @Description:
 */
class WifiReceiver : BaseReceiver() {
    companion object {
        private const val WIFI_ACTION = WifiManager.SCAN_RESULTS_AVAILABLE_ACTION
    }

    private var onScanCompletedListener: OnScanCompletedListener? = null

    interface OnScanCompletedListener {
        fun onScanCompleted(isSuccess: Boolean)
    }

    fun register(
        context: Context?,
        onScanCompleted: OnScanCompletedListener?
    ) {
        super.register(context, WIFI_ACTION)
        onScanCompletedListener = onScanCompleted
    }

    override fun onReceive(context: Context, intent: Intent) {
        var success = false
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            success = intent.getBooleanExtra(WifiManager.EXTRA_RESULTS_UPDATED, false)
        }
        val action = intent.action
        if (!TextUtils.isEmpty(action) && WIFI_ACTION == action) {
            onScanCompletedListener?.onScanCompleted(success)
        }
    }
}